consensus/state.go

```diff
@@ -1028,6 +1028,11 @@ func (cs *State) createProposalBlock() (block *types.Block, blockParts *types.Pa
 		cs.Logger.Error("enterPropose: Cannot propose anything: No commit for the previous block.")
 		return
 	}
+	if cs.privValidator == nil {
+		// should not happen
+		cs.Logger.Error("enterPropose: Cannot propose anything: validator keys are not present.")
```

Comment:
should we panic here since it's clearly our error?

```
func assertPrivValidatorExists() {
  if cs.privValidator == nil {
    panic("expected privValidator to exist")
  }
}
```

Reply:
this could also stem from kms/hsm going down. It makes sense to panic since it will mean it is a larger issue at hand.

Reply:
is it possible that the hsm would recover?

Reply:
Changed to panic. enterPropose ensures we don't call createProposalBlock if the node is not a validator & proposer in the current round.